sars2seq/translate.py

```diff
@@ -133,3 +147,27 @@ def translateSpike(seq):
             current += (subsequentGaps + index)
 
     return sequence
+
+
+def checkSpikeInsertions(accession, seq):
+    """
+    Check and parse out known insertions in the Spike protein.
+
+    @param seq: A C{str} amino acid sequence.
+    """
+    seqLen = len(seq)
+    if seqLen == 1274:
+        # There are no insertions
+        return seq
+    if seqLen > 1274:
+        if 'QTKGIALSPR' in seq:
+            return seq[:679] + seq[683:]
+        elif 'NLVRTDRDLPQ' in seq:
+            return seq[:214] + seq[217:]
+        elif 'LVRAAGYLPQ' in seq:
+            return seq[:214] + seq[217:]
```

Comment:
It's easy to add some tests for this, though maybe not needed. I would put in some that have the exact known AA sequences.

Also, I would guess you could make these more specific, by giving a start offset and using `find`. E.g.,

```python
if seq.find('QTKGIALSPR', 650, 700) > -1:
    return ...
```

Those (example) numbers delimit where in the sequence the pattern is looked for. This prevents the (unlikely, I guess?) scenario in which one of these AA sequences occurs earlier in the sequence.

Reply:
Done.